x/dex/keeper/core.go

```diff
@@ -806,6 +806,10 @@ func (k Keeper) WithdrawFilledLimitOrderCore(
 	if !found {
 		return types.ErrValidLimitOrderMapsNotFound
 	}
+	// checks that the user has some number of limit order shares wished to withdraw
+	if trancheUser.SharesOwned.LTE(sdk.ZeroInt()) {
```

Comment:
Does it make more sense to do sharesOwned - SharesCancelled here?

Reply:
Technically we only need to check if they have any cancelled shares since cancel is all or nothing now. But I think either is fine

Reply:
good catch